test/e2e/testsuites/testsuites.go

```diff
@@ -768,6 +770,166 @@ func waitForStatefulSetComplete(ctx context.Context, cs clientset.Interface, ns
+				Name:       volumeName,
+				DevicePath: mountPath,
+			},
+		}
+	}
+
+	testJob := &TestJob{
+		client:    c,
+		namespace: ns,
+		job: &batchv1.Job{
+			ObjectMeta: metav1.ObjectMeta{
+				GenerateName: generateName,
+			},
+			Spec: batchv1.JobSpec{
+				Template: v1.PodTemplateSpec{
+					ObjectMeta: metav1.ObjectMeta{
+						Labels: map[string]string{"app": selectorValue},
+					},
+					Spec: v1.PodSpec{
+						NodeSelector: map[string]string{"kubernetes.io/os": "linux"},
+						Containers: []v1.Container{
+							{
+								Name:          "volume-tester",
+								Image:         imageutils.GetE2EImage(imageutils.BusyBox),
+								Command:       []string{"/bin/sh"},
+								Args:          []string{"-c", command},
+								VolumeMounts:  volumeMounts,
+								VolumeDevices: volumeDevices,
+							},
+						},
+						RestartPolicy: v1.RestartPolicyNever,
+						Volumes: []v1.Volume{
+							{
+								Name: volumeName,
+								VolumeSource: v1.VolumeSource{
+									PersistentVolumeClaim: &v1.PersistentVolumeClaimVolumeSource{
+										ClaimName: pvc.Name,
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	if ttlSecondsAfterFinished != nil {
+		testJob.job.Spec.TTLSecondsAfterFinished = ttlSecondsAfterFinished
+	}
+
+	if useAntiAffinity {
+		affinity := &v1.Affinity{
+			PodAntiAffinity: &v1.PodAntiAffinity{
+				RequiredDuringSchedulingIgnoredDuringExecution: []v1.PodAffinityTerm{
+					{
+						LabelSelector: &metav1.LabelSelector{
+							MatchLabels: map[string]string{"app": selectorValue},
+						},
+						TopologyKey: driver.TopologyKey,
+					},
+				},
+			},
+		}
+		testJob.job.Spec.Template.Spec.Affinity = affinity
+	}
+
+	if isWindows {
+		testJob.job.Spec.Template.Spec.NodeSelector = map[string]string{
+			"kubernetes.io/os": "windows",
+		}
+		testJob.job.Spec.Template.Spec.Containers[0].Image = "mcr.microsoft.com/windows/servercore:" + getWinImageTag(winServerVer)
+		if useCMD {
+			testJob.job.Spec.Template.Spec.Containers[0].Command = []string{"cmd"}
+			testJob.job.Spec.Template.Spec.Containers[0].Args = []string{"/c", command}
+		} else {
+			testJob.job.Spec.Template.Spec.Containers[0].Command = []string{"powershell.exe"}
+			testJob.job.Spec.Template.Spec.Containers[0].Args = []string{"-Command", command}
+		}
+	}
+
+	return testJob
+}
+
+func (t *TestJob) Create(ctx context.Context) {
+	var err error
+	t.job, err = t.client.BatchV1().Jobs(t.namespace.Name).Create(ctx, t.job, metav1.CreateOptions{})
+	framework.ExpectNoError(err)
+}
+
+func (t *TestJob) WaitForAttachBatchCheck(ctx context.Context) error {
+	selector, err := metav1.LabelSelectorAsSelector(t.job.Spec.Selector)
+	framework.ExpectNoError(err)
+	jobPods, err := e2epod.WaitForPodsWithLabel(ctx, t.client, t.namespace.Name, selector)
+	framework.ExpectNoError(err)
+
+	err = waitForPodEvent(ctx, t.client, &jobPods.Items[0], "The maximum number of data disks allowed to be attached to a VM of this size is", 3*time.Minute)
```

Comment:
this issue was fixed by https://github.com/kubernetes-sigs/azuredisk-csi-driver/pull/3039, I don't think we need such test any more. @hccheng72 could you work out a PR only including the refine changes under pkg/azuredisk folder? thx

Reply:
yes, it shouldn't have `The maximum number of data disks allowed to be attached to a VM of this size is` (MaximumDataDisksExceeded) with #3039 but the test will succeed. The test monitors the scenario, if MaximumDataDisksExceeded is found during attaching process then we fail the test.

Reply:
@hccheng72  Have we checked if this test fails without the changes in  #3039 ?

Reply:
@nearora-msft Yes, if I revert #3039 fix, the test will fail.